### DISCUSSION TITLE
Version 0.6.0

### DISCUSSION THREAD

**Discussion by andsteing** [google/flax#2404] [General]
## What's Changed
* Add `on_commit_callback` to put the responsibility of renaming the directories on the users of the serialization library. This will also fix the GCS atomic rename issue where the users can write a success file when the commit is successful and check the existence of that file before deserialization. by @copybara-service in https://github.com/google/flax/pull/2328
* RDT Redesign by @cgarciae in https://github.com/google/flax/pull/2177
* Fix stale URLs to read the docs site. by @levskaya in https://github.com/google/flax/pull/2338
* Replace all jax.tree_* calls with jax.tree_util.tree_* by @levskaya in https://github.com/google/flax/pull/2325
* Further fix the singular-leaf checkpointing, and add tests. by @copybara-service in https://github.com/google/flax/pull/2336
* Forward unroll argument in scan_with_axes by @sanchit-gandhi in https://github.com/google/flax/pull/2339
* Document repo analytics by @cgarciae in https://github.com/google/flax/pull/2317
* Improve flax basics by @cgarciae in https://github.com/google/flax/pull/2291
* Split build into multiple jobs by @cgarciae in https://github.com/google/flax/pull/2277
* Fix type annotation for `step` in `training.checkpoints` by @Chuxiaof in https://github.com/google/flax/pull/2343
* Add test for writing and restoring empty checkpoints. by @copybara-service in https://github.com/google/flax/pull/2345
* Allow all processes to checkpoint when not using GDA by @copybara-service in https://github.com/google/flax/pull/2350
* Make importing tensorstore optional and move related type hints to comments. by @copybara-service in https://github.com/google/flax/pull/2348
* Use jax.named_scope for name stack rather than named_call. by @copybara-service in https://github.com/google/flax/pull/2349
* Internal change by @copybara-service in https://github.com/google/flax/pull/2356
* Fix sphinx CI errors by @cgarciae in https://github.com/google/flax/pull/2361
* Forward path to rewound Scope by @jheek in https://github.com/google/flax/pull/2360
* Make link a link on the getting started by @Davidnet in https://github.com/google/flax/pull/2340
* Fix colab & github links by @cgarciae in https://github.com/google/flax/pull/2363
* Fix ConvTranspose with circular padding by @cgarciae in https://github.com/google/flax/pull/2364
* Add some docstrings to the old flax.training.common_utils module. by @levskaya in https://github.com/google/flax/pull/2373
* Correct state variable name by @Jeevesh8 in https://github.com/google/flax/pull/2369
* Allow linen's Conv layer to operate on arbitrary-rank inputs. by @copybara-service in https://github.com/google/flax/pull/2308
* Copies `dynamic_scale.py` from `optim/` to `training/`. by @copybara-service in https://github.com/google/flax/pull/2375
* Add option `auto_flush` to `flax.metrics.tensorboard.SummaryWriter` by @copybara-service in https://github.com/google/flax/pull/2376
* updated supported transforms in lifting docs. by @levskaya in https://github.com/google/flax/pull/2374
* Test docstrings with autodoc on CI by @cgarciae in https://github.com/google/flax/pull/2372
* skip remat test that fails with autodiff by @mattjj in https://github.com/google/flax/pull/2389
* Removes `flax.optim.dynamic_scale`. by @copybara-service in https://github.com/google/flax/pull/2314
* Plumb spmd_axis_name from vmap_with_axes through to JAX vmap by @copybara-service in https://github.com/google/flax/pull/2390
* fixed math expressions  by @banda-larga in https://github.com/google/flax/pull/2392

## New Contributors
* @sanchit-gandhi made their first contribution in https://github.com/google/flax/pull/2339
* @Chuxiaof made their first contribution in https://github.com/google/flax/pull/2343
* @Davidnet made their first contribution in https://github.com/google/flax/pull/2340
* @Jeevesh8 made their first contribution in https://github.com/google/flax/pull/2369
* @banda-larga made their first contribution in https://github.com/google/flax/pull/2392

**Full Changelog**: https://github.com/google/flax/compare/v0.5.3...v0.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.6.0'>Version 0.6.0</a>.</em>
